distributed/scheduler.py

```diff
@@ -1768,7 +1781,9 @@ def transition_no_worker_memory(
                 pdb.set_trace()
             raise
 
-    def decide_worker(self, ts: TaskState) -> WorkerState | None:
+    def decide_worker(
+        self, ts: TaskState, recommendations: dict[str, str]
+    ) -> WorkerState | None:
```

Comment:
Good point. I'm going to look at how feasible it would be to move this question somewhere else. The tricky part is that it depends on the "is this a root-ish task" question, but maybe we can factor that out as well.